src/compiler.ts

```diff
@@ -7261,8 +7271,10 @@ export class Compiler extends DiagnosticEmitter {
     newExpr: ExpressionRef,
     /** Old value being replaced. */
     oldExpr: ExpressionRef,
-    /** The type shared between expressions. */
-    type: Type,
+    /** The type of the new expression. */
+    newType: Type,
+    /** The type of the old expression. */
+    oldType: Type,
```

Comment:
Suggesting to make overall parameter order here `newExpr, newType, oldExpr, oldType`.